### PR TITLE
Feat/checkbox tree

### PR DESCRIPTION
**Pull request by LimeWub**
JIRA: https://atomlearningltd.atlassian.net/browse/DS-494

Adds the full `CheckboxTree` component - which utilises `Tree` and `CheckboxGroup`, PRs for these two are open.

https://github.com/Atom-Learning/components/assets/6905473/b76c61a1-3374-413f-a837-520231c9bde4

### CODE REVIEW

**Review by MiguelGDLR — APPROVED** [review]

File: lib/src/components/checkbox-tree/CheckboxTreeItem.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  onCheckedChange,
+  children,
+  disabled,
+  ...rest
+}: CheckboxTreeItemProps): JSX.Element => {
+  const checkboxItemProps = { value, title, onCheckedChange, disabled }
+  return (
+    <Tree.Item {...rest}>
+      <CheckboxTreeItemContent>
+        <CheckboxGroup.Item {...checkboxItemProps} />
```
Comment (thomasdigby) [inline lib/src/components/checkbox-tree/CheckboxTreeItem.tsx:25]: Why not just pass these props directly?

File: documentation/content/components.form.primitives.checkbox-tree.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                <CheckboxTree.Item value={2}>Two</CheckboxTree.Item>
+                <CheckboxTree.Collapsible>
+                  <CheckboxTree.CollapsibleTrigger title="This is used for accessibility when someone hovers over/focuses the checkbox since it has no value for \`.AllItem\`." label="This is used for a11y when someone focuses the collapsable trigger."><Icon is={Person} />Trigger 1</CheckboxTree.CollapsibleTrigger>
+                  <CheckboxTree.CollapsibleContent>
+                    <CheckboxTree.Item value="nested 1"><Icon is={Person} />Nested: One</CheckboxTree.Item>
+                    <CheckboxTree.Item value={4}>Nested: Two</CheckboxTree.Item>
+                  </CheckboxTree.CollapsibleContent>
+                </CheckboxTree.Collapsible>
+                <CheckboxTree.Collapsible defaultOpen={true}>
+                  <CheckboxTree.CollapsibleTrigger title="trigger 2" label="Open: Trigger 2"><Icon is={Person} />Trigger 2 Trigger 2 Trigger 2 Trigger 2 Trigger 2 Trigger 2 Trigger 2 Trigger 2</CheckboxTree.CollapsibleTrigger>
```
Comment (thomasdigby) [inline documentation/content/components.form.primitives.checkbox-tree.md:35]: Can we replace this example trigger text with something more realistic? A subject/atom/topic checkbox tree could be more useful and would still retain the varying text length to show how the component reacts in that case

**Review by thomasdigby — APPROVED** [review]

**Review by Mhoog — APPROVED** [review]

File: lib/src/components/checkbox-tree/CheckboxTreeItem.tsx (hunk 1 above)
Comment (LimeWub) [reply lib/src/components/checkbox-tree/CheckboxTreeItem.tsx:25]: It's just a stylistic choice so it's clear to me which props are getting passed through to the checkbox specifically :) Can be directly passed too ^^

File: documentation/content/components.form.primitives.checkbox-tree.md (hunk 2 above)
Comment (LimeWub) [reply documentation/content/components.form.primitives.checkbox-tree.md:35]: I'm happy to replace the examples with actual Atom relevant data - but I thought it was fine to use more generic content to show functionality. This case shows the 3 dots (truncation) which I'm not sure we'd get with regular Atom data...

Comment (thomasdigby) [reply documentation/content/components.form.primitives.checkbox-tree.md:35]: Yeah I think generic content is fine, but `Trigger 2 Trigger 2 Trigger 2 Trigger 2 Trigger 2 Trigger 2 Trigger 2 Trigger 2` is just not realistic and particularly useful. You could go for Roles and users/names? You could definitely get some long names to trigger the truncation view. e.g.

```
[ ] Administrators
    [ ] Some long name
    [ ] Another long name
[ ] Another role
    [ ] Hubert Blaine Wolfeschlegelsteinhausenbergerdorff Sr.
```

Comment (LimeWub) [reply documentation/content/components.form.primitives.checkbox-tree.md:35]: Okay~ thanks for the suggestion, I'll replace it

File: lib/src/components/checkbox-tree/CheckboxTreeItem.tsx (hunk 1 above)
Comment (thomasdigby) [reply lib/src/components/checkbox-tree/CheckboxTreeItem.tsx:25]: I'm not sure what we gain here that we also lose by passing them directly... but hey ho #whatevs #noworries 👌

**Pull request merged**